mayastor/src/nexus_uri.rs

```diff
@@ -76,3 +82,42 @@ pub async fn bdev_destroy(uri: &str) -> Result<(), NexusBdevError> {
 pub fn bdev_get_name(uri: &str) -> Result<String, NexusBdevError> {
     Ok(Uri::parse(uri)?.get_name())
 }
+
+impl std::cmp::PartialEq<url::Url> for &Bdev {
+    fn eq(&self, uri: &url::Url) -> bool {
+        match Uri::parse(&uri.to_string()) {
+            Ok(device) => {
+                let uri_driver = if uri.scheme() == "nvmf" {
```

Comment:
it looks like where we converting  a nvmf scheme into nvme? why is that needed?

Reply:
because the driver name reported as nvme and not nvmf. I guess because it's indeed nvme but with a "fabrics" transport.